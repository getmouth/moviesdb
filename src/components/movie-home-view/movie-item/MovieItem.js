/* eslint-disable react/default-props-match-prop-types */
/* eslint-disable react/require-default-props */
/* eslint-disable camelcase */
/* eslint-disable react/destructuring-assignment */
// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { imageURL } from '../../../actions/URL';
import { fetchMovie } from '../../../actions/movie';
import './MovieItem.scss';

type Props = {
  movie: {
    id: number,
    title: string,
    release_date: string,
    vote_average: number,
    poster_path: string,
  },
  fetchMovie:(id: number) => void;
}

export const MovieItem = (props: Props) => {
  const {
    id,
    title,
    release_date,
    vote_average,
    poster_path,
  } = props.movie;

  let date;
  if (release_date !== undefined) {
    date = release_date.split('-');
  }

  return (
    <div
      role="button"
      tabIndex={0}
      className="MovieItem"
      onClick={() => props.fetchMovie(id)}
    >
      <div className="MovieImg">
        <Link to={`/movie/${id}-${title}`}>
          <img src={imageURL(poster_path)} alt={title} />
        </Link>
      </div>
      <div className="Title">
        <Link to={`/movie/${title}`}>
          <h2>{title}</h2>
        </Link>
      </div>
      <div className="MovieInfo">
        <div className="YearCol">
          <span>Year</span>
          {
            date !== null && <h3>{date[0]}</h3>
          }
        </div>
        <div className="RatingCol">
          <span>Rating</span>
          <h3>{vote_average}</h3>
        </div>
      </div>
    </div>
  );
};

const MapDispatchToProp = {
  fetchMovie,
};

MovieItem.defaultProps = {
  movie: {
    id: null,
    title: '',
    release_date: '',
    vote_average: null,
    poster_path: null,
  },
};

export default connect(null, MapDispatchToProp)(MovieItem);
