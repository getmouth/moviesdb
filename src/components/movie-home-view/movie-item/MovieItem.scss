$heading-color: rgba(0,0,0,0.6);
$subheading-color: rgba(0,0,0,0.5);

.MovieItem {
  width: 22.5%;
  box-shadow: 0px 1px 6px 0px rgba(0,0,0,0.2);
  margin-right: 25px;
  margin-bottom: 30px;
  display: flex;
  flex-direction: column;
  &:nth-child(4n + 0) {
    margin-right: 0;
  }

  .Title {
    h2 {
      font-size: 18px;
    }

    a {
      text-decoration: none;
      color: $heading-color;
    }
    padding: 0 10px;
  }

  .MovieImg {

    width: 100%;
    img {
      width: 100%;
      max-height: 300px;
    }
  }

 /***********************************
   Information Part of the movie item/card component
 ************************************/
  .MovieInfo {
    margin-top: auto;
    padding: 0 10px;
    display: flex;
    justify-content: space-between;


    .YearCol, .RatingCol {

      span {
        font-size: 10px;
        margin-bottom: 0px;
        color: $subheading-color;
      }
      h3 {
        color: $heading-color;
        font-size: 12px;
        font-weight: bold;
        margin: 8px 0px;
      }

    }

    .YearCol {
      text-align: left;
      align-self: flex-start;
    }

    .RatingCol {
      text-align: right;
      align-self: flex-end;
    }

  }

  /***********************************
   Media Query BreakPoint for Display sizes
 ************************************/

  @media all and (max-width: 1024px) {
    width: 29%;
    &:nth-child(3n + 0) {
      margin-right: 0px;
    }

    &:nth-child(4n + 0) {
      margin-right: 25px;
    }

    .Title {
      h2 {
        font-size: 16px;
      }
    }


  }

  @media all and (max-width: 970px) {
    width: 28%;
    &:nth-child(3n + 0) {
      margin-right: 0px;
    }

    &:nth-child(4n + 0) {
      margin-right: 25px;
    }

    .Title {
      h2 {
        font-size: 14px;
      }
    }
  }

  @media all and (max-width: 820px) {
    width: 40%;
    margin-right: 30px;
    &:nth-child(2n + 0) {
      margin-right: 0px;
    }

    &:nth-child(3n + 0) {
      margin-right: 30px;
    }
  }

  @media all and (max-width: 768px) {
    width: 35%;
    margin-right: 30px;
    &:nth-child(2n + 0) {
      margin-right: 0px;
    }


    .MovieImg {
      width: 100%;

      img {
        width: 100%;
        max-height: 400px;
      }
    }
  }

  @media all and (max-width: 600px) {
    width: 60%;
    margin-right: 0;
    margin-bottom: 40px;
    &:nth-child(3n + 0) {
      margin-right: 0px;
    }
  }

  @media all and (max-width: 480px) {
    width: 80%;
  }
}
