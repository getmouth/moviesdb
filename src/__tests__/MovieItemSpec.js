import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { MovieItem } from '../components/movie-home-view/movie-item/MovieItem';

describe('<MovieItem />', () => {
  it('Should not Break When no Movie is passed', () => {
    const component = renderer.create(<BrowserRouter><MovieItem /></BrowserRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should Correctly Render When Movie is passed', () => {
    const movie = {
      id: 1,
      title: 'test movie',
      release_date: '2019-10-10',
      vote_averge: 9.4,
      poster_path: null,
    };

    const component = renderer.create(<BrowserRouter><MovieItem movie={movie} /></BrowserRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
