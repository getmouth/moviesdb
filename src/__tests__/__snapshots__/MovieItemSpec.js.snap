// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem /> Should Correctly Render When Movie is passed 1`] = `
<div
  className="MovieItem"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MovieImg"
  >
    <a
      href="/movie/test movie"
      onClick={[Function]}
    >
      <img
        alt="test movie"
        src="https://image.tmdb.org/t/p/originalnull"
      />
    </a>
  </div>
  <div
    className="Title"
  >
    <a
      href="/movie/test movie"
      onClick={[Function]}
    >
      <h2>
        test movie
      </h2>
    </a>
  </div>
  <div
    className="MovieInfo"
  >
    <div
      className="YearCol"
    >
      <span>
        Year
      </span>
      <h3>
        2019
      </h3>
    </div>
    <div
      className="RatingCol"
    >
      <span>
        Rating
      </span>
      <h3 />
    </div>
  </div>
</div>
`;

exports[`<MovieItem /> Should not Break When no Movie is passed 1`] = `
<div
  className="MovieItem"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MovieImg"
  >
    <a
      href="/movie/"
      onClick={[Function]}
    >
      <img
        alt=""
        src="https://image.tmdb.org/t/p/originalnull"
      />
    </a>
  </div>
  <div
    className="Title"
  >
    <a
      href="/movie/"
      onClick={[Function]}
    >
      <h2>
        
      </h2>
    </a>
  </div>
  <div
    className="MovieInfo"
  >
    <div
      className="YearCol"
    >
      <span>
        Year
      </span>
      <h3>
        
      </h3>
    </div>
    <div
      className="RatingCol"
    >
      <span>
        Rating
      </span>
      <h3 />
    </div>
  </div>
</div>
`;
